tests/Fake/DataObjectFakeVersioned.php

```diff
@@ -13,7 +14,7 @@
  * @mixin SearchServiceExtension
  * @mixin Versioned
  */
-class DataObjectFakeVersioned extends DataObject implements TestOnly
+class DataObjectFakeVersioned extends DataObject implements TestOnly, DocumentInterface
```

Comment:
the DocumentInterface is intended for things that are directly passed to the indexer which for standard DataObjects is the DataObjectDocument class - adding the interface here is muddying the waters a bit. I see what you're trying to do here but as it stands I think this is mainly just testing that you've set 'ShowInSearch' to some value as this class will simply pass that value back.

Instead, you'll need to create a new DataObjectDocument and pass this DataObjectFakeVersioned into it (theres a few examples of this in the the same test). You will need to publish it and meet the other conditions in the DataObjectDocument::shouldIndex method.

Reply:
Seemed liked I was over engineering it. Updated code.